src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -155,20 +201,21 @@ void parseXml(List<CastVoteRecord> castVoteRecords)
       HashMap<String, Candidate> candidateById = new HashMap<>();
       for (Election election : cvrReport.Election) {
         for (Candidate candidate : election.Candidate) {
-          candidateById.put(candidate.ObjectId, candidate);
+          putIfUnique(candidateById, candidate.ObjectId, candidate, "Candidate");
```

Comment:
Similar to my above comment, the way I like to design validation flows typically is to have a boolean that gets triggered if any conditions fail, but only check that boolean at the very end all the validation and throw the exception at that point. This allows us to avoid situations where there are maybe 30 things wrong with an input, but we don't need to run it 30 times (each after fixing the one error it gives us) to figure every one of these things out. As is, this helper throws the exception, so it doesn't behave that way.

Let me know if that doesn't make sense, or if you disagree.